static/src/javascripts/projects/commercial/modules/reload-page-on-consent-change.ts

```diff
@@ -0,0 +1,26 @@
+import { onConsentChange } from '@guardian/consent-management-platform';
+import type { ConsentState } from '@guardian/consent-management-platform/dist/types';
+import { once } from 'lodash-es';
+
+let initialConsentState: ConsentState | undefined;
+
+/**
+ * If consent has been set, and if consent then changes, reload the page so the correct
+ * state is reflected in the rendered page
+ */
+const _reloadPageOnConsentChange = (): void => {
+	onConsentChange((consent) => {
```

Comment:
is there a chance this could happen before other `onConsentChange` stuff happens? like setting/unsetting the ad free cookie/localStorage for example

Reply:
Really good question. I don't think we'd considered this. 
`reloadPageOnConsentChange` comes after `manageAdFreeCookieOnConsentChange` in `commercialExtraModules` and these functions are executed in order, each passing a callback to `onConsentChange`. Since `onConsentChange` executes its callbacks [in the order that they're registered in](https://github.com/guardian/consent-management-platform/blob/17db77bed362e99b25bb709bf33d5db06909c8a4/src/onConsentChange.ts#L105), I think we're guaranteed to set / unset the ad free cookie before reloading.
But relying on `commercialExtraModules`’s current order seems a bit precarious. We could add a comment explaining the importance of the ordering and / or do `manageAdFreeCookieOnConsentChange.then(reloadPageOnConsentChange)` so that it's clearer that order matters. 
This wouldn’t protect us from `reloadPageOnConsentChange` reloading the page before some other uses of `onConsentChange` have a chance to have their callbacks executed. As far as I can tell, at the moment all uses of `onConsentChange` apart from `reloadPageOnConsentChange` and `manageAdFreeCookieOnConsentChange` only really care about and do anything with the _initial_ consent state and disregard subsequent changes. Even so, it seems a shame to introduce the side effect of maybe swallowing other callbacks based on when they happen to be registered.

Reply:
`commercialExtraModules` is loaded in parallel as far as I can see? https://github.com/guardian/frontend/blob/main/static/src/javascripts/bootstraps/standalone.commercial.ts#L208

The only non parallel loading is of permutive before googletag I think.

Reply:
Yes we do a `Promise.all()` on the array of promises returned by the commercial module functions but since they're [called in order](https://github.com/guardian/frontend/blob/main/static/src/javascripts/bootstraps/standalone.commercial.ts#L137) I think `manageAdFreeCookieOnConsentChange`'s callback should be pushed onto `onConsentChange`'s queue before `manageAdFreeCookieOnConsentChange`.
We discussed this at the last comm dev time and decided that instead of relying on the order of execution of the commercial modules' init functions, we should instead change `onConsentChange`'s API so that certain callbacks can be guaranteed to run last. I've had a go at this in [this PR](https://github.com/guardian/consent-management-platform/pull/664).